service/src/main/java/bio/terra/workspace/common/utils/FlightUtils.java

```diff
@@ -277,4 +280,17 @@ public static boolean flightComplete(FlightState flightState) {
         || flightState.getFlightStatus() == FlightStatus.FATAL
         || flightState.getFlightStatus() == FlightStatus.SUCCESS);
   }
+
+  public static SamUser getRequiredSamUser(FlightMap inputParameters, SamService samService) {
+    AuthenticatedUserRequest userRequest =
+        FlightUtils.getRequired(
+            inputParameters,
+            JobMapKeys.AUTH_USER_INFO.getKeyName(),
+            AuthenticatedUserRequest.class);
+    return samService.getSamUser(userRequest);
+  }
+
+  public static String getRequiredUserEmail(FlightMap inputParameters, SamService samService) {
```

Comment:
This might be getting called multiple times per flight. Would it make sense to store it in the working map instead?

Reply:
Agree, want to make a single call. 
I will create a PR for this change + removing the redundant user request in create flow (see John's comment). 
Reduced the scope of this PR and reverted my previous changes which fetched the user from sam prior to the flight

Reply:
tracking - https://verily.atlassian.net/browse/BENCH-1097